tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/typebynode/TypeByMiscExprTest.java

```diff
@@ -0,0 +1,132 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.semantic.api.test.typebynode;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.BinaryExpressionNode;
+import io.ballerina.compiler.syntax.tree.ConditionalExpressionNode;
+import io.ballerina.compiler.syntax.tree.LetExpressionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.TypeTestExpressionNode;
+import io.ballerina.compiler.syntax.tree.UnaryExpressionNode;
+import io.ballerina.compiler.syntax.tree.XMLFilterExpressionNode;
+import io.ballerina.compiler.syntax.tree.XMLStepExpressionNode;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.ballerina.compiler.api.symbols.TypeDescKind.BOOLEAN;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.INT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.OBJECT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.STRING;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.XML;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests for getting the type of misc. exprs.
+ *
+ * @since 2.0.0
+ */
+@Test
+public class TypeByMiscExprTest extends TypeByNodeTest {
+
+    @Override
+    String getTestSourcePath() {
+        return "test-src/type-by-node/type_by_misc_exprs.bal";
+    }
+
+    @Override
+    NodeVisitor getNodeVisitor(SemanticModel model) {
+        return new NodeVisitor() {
+            @Override
+            public void visit(UnaryExpressionNode unaryExpressionNode) {
+                assertType(unaryExpressionNode, model, INT);
+            }
+
+            @Override
+            public void visit(BinaryExpressionNode binaryExpressionNode) {
+                TypeDescKind typeKind;
+                switch (binaryExpressionNode.operator().text()) {
+                    case "-":
+                    case "*":
+                    case "+":
+                    case "/":
+                        typeKind = INT;
+                        break;
+                    case ">=":
+                    case "!=":
+                    case "==":
+                    case "&&":
+                        typeKind = BOOLEAN;
+                        break;
+                    case "...":
```

Comment:
Do we have constants/enums for these values?

Reply:
Ah yes. Will replace with the enum